src/handlers/miner_hbbft_handler.erl

```diff
@@ -529,5 +577,42 @@ md_version(Ledger) ->
             tuple
     end.
 
+%% do nothing if we've advanced
+process_skips(Proposed, SenderRound, F, Sender, Votes) ->
+    Votes1 = Votes#{Sender => {Proposed, SenderRound}},
+    PropVotes = maps:fold(fun(_Sender, {Vote, _OwnRound}, Acc) when Vote =:= Proposed ->
+                                  Acc + 1;
+                             (_Sender, {_OtherVote, _OwnRound}, Acc) ->
+                                  Acc
+                          end,
+                          0,
+                          Votes1),
+    %% equal here because we only want to go once, at the point of saturation
+    case PropVotes == (2 * F) + 1 of
+        true -> {skip, Votes1};
+        false -> {wait, Votes1}
+    end.
+
+%% the idea here is to take a clean round that's higher than the median, which should be relatively
+%% hard to manipulate by cheating
+median_not_taken(Map) ->
+    {_Votes, Rounds} = lists:unzip(maps:values(Map)),
+    lager:info("rounds ~p", [lists:sort(Rounds)]),
+    Median = miner_util:median(Rounds),
+    search_lowest(Median, lists:sort(Rounds)).
+
+%% if we haven't found a round in the list, the next highest should work
+search_lowest(Try, []) ->
```

Comment:
Can we add a simple eunit for this particular function?

Reply:
Also a spec.